protocol/avdl/keybase1/teams.avdl

```diff
@@ -1621,31 +1621,53 @@ protocol teams {
 
   AnnotatedTeam getAnnotatedTeam(TeamID teamID);
 
-  record AnnotatedSubteamMemberDetails {
-    TeamName teamName;
-    TeamID teamID;
-    TeamMemberDetails details;
-    TeamRole role;
-  }
-  array<AnnotatedSubteamMemberDetails> getUserSubteamMemberships(TeamID teamID, string username);
-
   // For member page
   record TeamTreeMembershipValue {
     TeamRole role;
     // last time the user joined the team
     union { null, Time } joinTime;
+    TeamID teamID;
   }
   enum TeamTreeMembershipStatus {
     OK_0,
-    ERROR_1
+    ERROR_1,
+    HIDDEN_2
+  }
+  record TeamTreeError {
+    string message;
+    // willSkipSubtree is true if loading of the subtree rooted at the team associated
+    // with this error was cancelled due to this error.
+    // is reached loading an ancestor team.
+    boolean willSkipSubtree;
+    // willSkipAncestors is true if loading of the ancestor chain of the team associated
+    // with this error was cancelled due to this error.
+    boolean willSkipAncestors;
   }
   variant TeamTreeMembershipResult switch (TeamTreeMembershipStatus s) {
     case OK: TeamTreeMembershipValue;
-    case ERROR: GenericError;
+    case ERROR: TeamTreeError;
+    // HIDDEN is used for membership results of teams the user is not in
+    case HIDDEN: void;
   }
   record TeamTreeMembership {
-    TeamName teamName;
+    string teamName;
     TeamTreeMembershipResult result;
+
+    // bookkeeping params; 'target' refers to the initial request
+    TeamID targetTeamID;
+    string targetUsername;
+    int guid;
+  }
+  record TeamTreeMembershipsDoneResult {
+    int expectedCount;
+
+    // bookkeeping params; 'target' refers to the initial request
+    TeamID targetTeamID;
+    string targetUsername;
+    int guid;
+  }
+  record TeamTreeInitial {
```

Comment:
It's set in the gui's engine layer and not exposed (yet) to the main codepaths.

Reply:
this whole sessionId thing feels a bit hacky, as if it wasn't intended for this purpose. Another option would be just sample a random GUID and forget the sessionId, so that this doesn't break if we change the rpc layer. Again, up to you.

Reply:
I talked to Patrick and he said this was a good use. It's just that there aren't any major uses of the sessionID in the gui yet.